### PR TITLE
Add `@starting-style` variant

### PR DESCRIPTION
**Pull request by nix6839**
This PR adds support for `@starting-style` through the `starting:` variant.

`@starting-style` was [shipped in Chrome 117](https://chromestatus.com/feature/4515377717968896)

Further Reading:
- https://www.bram.us/2023/05/24/the-yellow-fade-technique-with-modern-css-using-starting-style/

### CODE REVIEW

File: src/lib/setupContextUtils.js [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
-import negateValue from '../util/negateValue'
-import isSyntacticallyValidPropertyValue from '../util/isSyntacticallyValidPropertyValue'
-import { generateRules, getClassNameFromSelector } from './generateRules'
+import transformThemeValue from '../util/transformThemeValue'
 import { hasContentChanged } from './cacheInvalidation.js'
+import { generateRules, getClassNameFromSelector } from './generateRules'
 import { Offsets } from './offsets.js'
-import { finalizeSelector, formatVariantSelector } from '../util/formatVariantSelector'
+import * as sharedState from './sharedState'
+import { env } from './sharedState'
```
Comment (RobinMalfait) [inline src/lib/setupContextUtils.js:24]: Hey, please revert these changes. There's no need to change them.

Comment (nix6839) [reply src/lib/setupContextUtils.js]: Srry, my mistake

**Comment by lukewarlow** [comment]
See https://developer.chrome.com/blog/entry-exit-animations/ for a good overview on all these new capabilities in Chrome

**Comment by lukewarlow** [comment]
https://github.com/WebKit/WebKit/pull/23501 - this will soon be supported in WebKit also.

**Comment by adamwathan** [comment]
Thanks! Going to target v4 for this one, have opened a new PR and included you as a co-author here: https://github.com/tailwindlabs/tailwindcss/pull/12040
